base58/Cargo.toml

```diff
@@ -0,0 +1,27 @@
+[package]
+name = "bitcoin-base58"
+version = "0.1.0"
+authors = ["Andrew Poelstra <apoelstra@wpsoftware.net>", "The Rust Bitcoin developers"]
+license = "CC0-1.0"
+repository = "https://github.com/rust-bitcoin/rust-bitcoin/"
+documentation = "https://docs.rs/bitcoin-base58"
+description = "Bitcoin base58 encoding format"
+categories = ["cryptography::cryptocurrencies"]
+keywords = ["bitcoin", "encoding"]
+readme = "README.md"
+edition = "2018"
+
+# Please don't forget to add relevant features to docs.rs below.
+[features]
+default = ["std"]
+std = ["bitcoin_hashes/std"]
```

Comment:
`std` should depend on `alloc`

Reply:
I thought that was the case, but `bitcoin_hashes` does not do it like that - is it wrong? It looks like something to do with how we use `core2`.

Reply:
It is definitely something I'd like to fix. Both are usable and it's even required to *not* do that for crates with MSRV lower than the Rust version that introduced the `alloc` crate (`serde` is an example) but it's easier to work with it being written once in `Cargo.toml` than using `any(feature = "alloc", feature = "std")` all over the place.